llvm/include/llvm/IR/DataLayout.h

```diff
@@ -91,6 +91,9 @@ class DataLayout {
 private:
   bool BigEndian = false;
 
+  /// The size of a byte in bits.
+  unsigned ByteWidth = 8;
```

Comment:
I think we should store this as a shift value instead of a width since it will allow Clang/GCC to elide divisions (see https://godbolt.org/z/5T86s84Ez case 3 optimizes to a right shift).
And it looks like this also works for divideCeil: https://godbolt.org/z/1oa18Gns7

Reply:
I thought about it too, but decided to leave it as is until there is an evidence that it affects compilation time. I didn't measure yet.

Reply:
Forcing everyone to deal with a shift value is not very ergonomic, so to be clear I'm not suggesting using `getByteShift()`, but rather the approach of storing a shift value and then modifying `getByteSize()` to return `1u << shift`.

This will allow using uint8_t to fit it into existing padding and more importantly allow the compiler to elide divisions after inlining.